arch-snapshot/arch.md

```diff
@@ -50,36 +50,39 @@ details and establish a glossary to ensure consistency within the document.
+a suitable Offer, they contact the Provider directly to negotiate the deal.
+
+Typically, humans are not involved in the process of finding, matching, negotiating, or finalizing agreements. Instead,
+users define their needs programmatically, allowing the [Provider Agent](#provider-agent) and [Requestor Agent](#requester-agent)
+software to handle these tasks automatically.
 
 It is the main goal of the [Provider Agent](#provider-agent) to implement logic for selling resources
-in Golem Network. From high level perspective, Agent application should do following things: 
+in Golem Network. From high level perspective, Provider Agent application should do following things: 
 1. Describe Resources using property language to create an [Offer](#offer)
 2. Publish the Offer in the market
-3. Monitor incoming Proposal events and negotiate an [Agreement](#agreement) with the most promising [Requestor](#requestor)
+3. Monitor incoming Proposals and negotiate an [Agreement](#agreement) with the most promising [Requestor](#requestor)
 4. Allocate the promised Resources in accordance with the [Agreement](#agreement)
-5. Send [DebitNotes](#debit-note) to notify [Requestor](#requestor) of updated costs
+5. Monitor resources usage and charge Requestor Agent
 6. Terminate the Agreement or await the Agreement termination event from the [Requestor](#requestor)
 7. Send an [Invoice](#invoice) summarizing the total cost of the [Agreement](#agreement)
-8. Monitor Payment API events for [Invoice](#invoice) settlement and payment confirmation 
+8. Wait until the payment for the [Invoice](#invoice) is settled and payment confirmed. 
 
 #### 1. Describe Resources using property language to create an [Offer](#offer)
 
-The Golem design was created to support the sale of any type of computing resource.To achieve this,
-Golem employs a generic [property and constraints language](#discovery-and-offersdemand-matching) to
-describe the resources being offered. 
+While Golem is currently used for trading computational resources, it was designed to support the exchange of any type of
+[resource](#resource). This means the Marketplace does not enforce strict standards on the goods being traded.
+To enable this flexibility, Golem uses a generic [property and constraints language](#discovery-and-offersdemand-matching)
+to describe the resources being offered.
 
-The [Yagna market](#market) is agnostic to the specific properties used and can match Offers and Demands as long
-as they adhere to the language specification. However, the Yagna daemon does not interpret the semantics
-of the properties in the Offer, nor does its behavior depend on the negotiated Agreement. It is the
-responsibility of the Provider Agent application to accurately interpret the semantics and implement
+The Golem Node is agnostic to the specific properties used and can match Offers and Demands as long
```

Comment:
Why do we need to introduce the distinction between the "Golem Node" and "Provider Agent" here? The reason why I'm so stubborn to defer it to the next sections is because the reader has no idea what it is so it requires explanation, which IMO would only lower the readability of this section.

Reply:
Can we speak of the Golem Network instead of the Golem Node at least? It should still parse and remain factual.